charms/knative-serving/src/image_management.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+#
+# Learn more at: https://juju.is/docs/sdk
+from typing import Dict
+
+import yaml
+
+
+def parse_image_config(image_config: str) -> Dict[str, str]:
+    """Parses config data for a dict of images, returning the parsed value as a dict."""
+    try:
+        images_from_config = yaml.safe_load(image_config)
+    except yaml.YAMLError as err:
+        raise ValueError(
```

Comment:
I'm okay with raising `ValueError` here as, according to the [docs](https://docs.python.org/3/library/exceptions.html#ValueError): "Raised when an operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as [IndexError](https://docs.python.org/3/library/exceptions.html#IndexError).", so I could agree that if the `yaml` is not formatted correctly we could say it is a value error when `YAMLError` is raised, but it took me a while to understand why this is the case. Probably it would be more clear if the docstring of this method stated in which situation a `yaml.YAMLError` is raised and why we want to raise back a `ValueError` from that other exception.

Also, this is completely optional, but I believe `YAMLError` has information about the line and column where it found the parsing error.

Reply:
yeah I wasn't sure what was best here.  I wanted to simplify/add context for the error so the message could be better caught for a status, but it didn't quite feel right either.  

I've updated the docstrings, but give another read and see if it works for you.  I'm open to suggestions.  I like this as is if we're also logging the full error message (including the YAMLError, which like you say has more specific info).  But if we just catch to set a status, it feels opaque

Reply:
Maybe I should just let the YAMLError go uncaught, and just catch it in the charm and make a blocked message there instead.  wdyt?

Reply:
Yeah, I think that last option is good.

Reply:
Updated.  wdyt?